sandbox/libs/authn/src/main/java/org/opensearch/authn/AbstractPermission.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.authn;
+
+/**
+ * This is an abstract class that defines the minimum expectations for a permission object.
+ * A permission implementation of this class can implement further functionality or structuring.
+ * A permission needs to entail the action it allows and the resource its performed on.
+ */
+abstract class AbstractPermission {
```

Comment:
I think we should delete this class.  I think it makes sense to have a single representation of permission, I'm not seeing much value in extensibility, is there a scenario where we would want to use Abstract permission?

Reply:
I agree. I was not sure if we would want to allow the extension of different types of permissions for the future but I do not know what any other type of permission would look like.